### ISSUE TITLE
Action buttons in chooser components are cut off by overflow in 5.0

### ISSUE THREAD

**Issue by drimacus182** [wagtail/wagtail#10553]
### Issue Summary

Take Image chooser. Starting with middle screen widths action buttons become cut off. This is especially bad for image chooser component, because it displays image title now, with a quite large font. And the image title is often just a filename, not human readable, doesn't contain spaces. And it's even worse for panels with two column layout.

See the screenshot below:
![image](https://github.com/wagtail/wagtail/assets/3090883/5c1e4620-1d08-4567-9b35-6cf1cdbae36d)

### Steps to Reproduce

I'm using a fresh [bakerydemo](https://github.com/wagtail/bakerydemo) instance in gitpod (commit 046eb8ecf866b88b6e4f402a66e04eefea93f504)

1. Log in as admin
2. Open **Pages** -> **Blog** ->  **Desserts with Benefits** for edit
3. Scroll to the first image. You can also edit the image and make its title longer and without spaces (as on screenshot above). 
4. On the blog edit page shrink the browser window and observe action buttons hiding

As a simple fix I see adding `flex-wrap: wrap` to `.chooser__actions` block here:
https://github.com/wagtail/wagtail/blob/2a38248c6b9f99369d5086a8a9a2387e54038582/client/scss/components/_chooser.scss#L71-L73
But I'm not sure how it will affect other components. Also I feel that image title can be made smaller and wrappable, because it's not very important piece of information.

- I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: Yes, on a fresh bakerydemo in gitpod

### Technical details

- Python version: 3.8.12
- Django version: 4.2.2
- Wagtail version: 5.0.1
- Browser version: Firefox 113

**Comment by lb-** [comment]
This might be fixed by the new chooser actions redesign #10557

FYI @thibaudcolas

**Comment by Morsey187** [comment]
Would just like to flag that this issue can be confusing when it overlaps a delete button on a page, as the button can be hidden seamlessly (depending on the browser viewport) with no indication that the buttons are in fact behind the panel. Especially puzzling for new content edtiors less familiar with where the buttons are located and this quirk of the side panel overlapping.

<details>
  <summary>Panel hiding the delete buttons</summary>
<img width="466" alt="Screenshot 2024-02-09 at 14 47 20" src="https://github.com/wagtail/wagtail/assets/35934884/068f71e9-aa61-464a-ba13-3bbed2c340ce">
</details>

<details>
  <summary>Panel showing the delete buttons</summary>
<img width="437" alt="Screenshot 2024-02-09 at 14 47 25" src="https://github.com/wagtail/wagtail/assets/35934884/a3eee021-d7cd-4666-9d35-636136654102">
</details>


I'll follow up with the acessability team and see if I can get some design feedback on a quick solution, otherwise I think this will require some more thinking as I'm not sure we'd be able to squash the main form content (panel fields) dynamically based of the side bar without some major refactoring.

**Comment by thibaudcolas** [comment]
Tagging this as an accessibility issue because this must be particularly problematic for people who use browser or font size zoom.

@Morsey187 I think this is fixable without a redesign. Right now part of the issue is that tab panels’ width within the form is defined based on the full page width, without taking into account the side of the sidebar, side panel, or minimap. I think it’d be worthwhile to try and change that calculation in the CSS to take those sizes into account.

Or possibly give a go to [container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries).

**Comment by Morsey187** [comment]
Thanks for sharing your insights on this Thibuad, that's helpful to know, I'll take a look.

**Comment by lb-** [comment]
Some more recent work was done for the edit container widths when the side panels are open - https://github.com/wagtail/wagtail/pull/11150 / https://github.com/wagtail/wagtail/issues/11038

Might be worth looking at these as it's probably a similar fix for the Minimap overlapping. These changes had to consider multiple scenarios of side panel opening.

**Comment by Morsey187** [comment]
Thanks for sharing this LB, 
I've tested the changes in https://github.com/wagtail/wagtail/pull/11150 and this looks to resolve my concerns (same as the resolved issue 11038) as well as the inital issue described here.

I reckon it should be fine to close this issue with the solution being to upgrade to Wagtail 6.0.

**Comment by lb-** [comment]
I will close this as fixed in 6.0 - if there are similar issues encountered, please add a comment or raise a more specific bug report.

Thanks all.
